miraclelinux/meta-debian
# tzdata: Update version from 2021a-0+deb10u10 to 2021a-0+deb10u11

# Purpose of pull request
This version updated timezone database for Lebanon and leap second list.

DLA: DLA 3412-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Check the files installed by the tzdata package
4. Check the TZ string for each country changed by this version

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " tzdata"
```
```
$ bitbake core-image-minimal
```

### Run qemu image
```
$ runqemu qemuarm64 nographic
```

### Check the files installed by the tzdata package
```
# cd /usr/share/zoneinfo/
# ls -F
```

### Check the TZ string for each country changed by this version
Find the country code by look up the country name from iso3166.tab.
```
# grep -n '<country name>' /usr/share/zoneinfo/*.tab
```
Find the timezone file by look up the country code from zone.tab.
```
# grep -n '^<country code>' /usr/share/zoneinfo/*.tab
```
Output the TZ string of last line in timezone file.
```
# tail -n 1 <timezone file>
```

## Test result
### Build package
Build succeeded.
```
$ bitbake tzdata
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |########################################################################################################################| Time: 0:00:54
Parsing of 1362 .bb files complete (0 cached, 1362 parsed). 2384 targets, 72 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-20.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-tzdata_2021a-0+deb10u11:1737f49a3f34300dcd7db494ae33685cd932d4fd"
meta-debian-extended = "HEAD:e5c9f9bc2b7104f4ee89360eaeee3e96749960f4"
meta-emlinux         = "HEAD:946bc9498f07741e667ec2b670ad346277d6be0f"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |#####################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 52 Found 0 Missed 52 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 462 tasks of which 0 didn't need to be rerun and all succeeded.
```

### Package test
The tzdata package are installed.
```
root@qemuarm64:~# cd /usr/share/zoneinfo/
root@qemuarm64:/usr/share/zoneinfo# ls -F
Africa/       Brazil/       EST           GB            HST           Japan         NZ            Portugal      US/           posix/
America/      CET           EST5EDT       GB-Eire       Hongkong      Kwajalein     NZ-CHAT       ROC           UTC           right/
Antarctica/   CST6CDT       Egypt         GMT           Iceland       Libya         Navajo        ROK           Universal     zone.tab
Arctic/       Canada/       Eire          GMT+0         Indian/       MET           PRC           Singapore     W-SU          zone1970.tab
Asia/         Chile/        Etc/          GMT-0         Iran          MST           PST8PDT       SystemV/      WET
Atlantic/     Cuba          Europe/       GMT0          Israel        MST7MDT       Pacific/      Turkey        Zulu
Australia/    EET           Factory       Greenwich     Jamaica       Mexico/       Poland        UCT           iso3166.tab
```

Checked the TZ string for each country changed by this version.

Lebanon
```
root@qemuarm64:/usr/share/zoneinfo# grep -n 'Lebanon' /usr/share/zoneinfo/*.tab
/usr/share/zoneinfo/iso3166.tab:152:LB	Lebanon
root@qemuarm64:/usr/share/zoneinfo# grep -n '^LB' /usr/share/zoneinfo/*.tab
/usr/share/zoneinfo/iso3166.tab:152:LB	Lebanon
/usr/share/zoneinfo/zone.tab:247:LB	+3353+03530	Asia/Beirut
/usr/share/zoneinfo/zone1970.tab:218:LB	+3353+03530	Asia/Beirut
root@qemuarm64:/usr/share/zoneinfo# find . -name 'Lebanon'
root@qemuarm64:/usr/share/zoneinfo# tail -n 1 Asia/Beirut
EET-2EEST,M3.5.0/0,M10.5.0/0
```
